pages/404.js

```diff
@@ -1,6 +1,13 @@
-import React from "react";
 import Link from "next/link";
+import React, { useEffect, useState } from "react";
 function PageNotFound() {
+  const [preview, setPreview] = useState(null);
+
+  useEffect(() => {
+    const status = localStorage.getItem("preview");
```

Comment:
Based on suggested review from `/pages/[slug].js` and `/pages/index.js` can we check if we can pass the `preview` value to this function or even via separate Component that we can call here?